Java/examples/src/main/java/com/amazon/randomcutforest/examples/serialization/ProtostuffExampleWithDynamicLambda.java

```diff
@@ -0,0 +1,151 @@
+import io.protostuff.runtime.RuntimeSchema;
+
+/**
+ * Serialize a Random Cut Forest using the
+ * <a href="https://github.com/protostuff/protostuff">protostuff</a> library.
+ */
+public class ProtostuffExampleWithDynamicLambda implements Example {
+    public static void main(String[] args) throws Exception {
+        new ProtostuffExampleWithDynamicLambda().run();
+    }
+
+    @Override
+    public String command() {
+        return "protostuff";
+    }
+
+    @Override
+    public String description() {
+        return "serialize a Random Cut Forest with the protostuff library";
+    }
+
+    @Override
+    public void run() throws Exception {
+        // Create and populate a random cut forest
+
+        int dimensions = 4;
+        int numberOfTrees = 50;
+        int sampleSize = 256;
+        Precision precision = Precision.DOUBLE;
+
+        RandomCutForest forest = RandomCutForest.builder().compactEnabled(true).dimensions(dimensions)
+                .numberOfTrees(numberOfTrees).sampleSize(sampleSize).precision(precision).build();
+
+        int dataSize = 4 * sampleSize;
+        NormalMixtureTestData testData = new NormalMixtureTestData();
+        for (double[] point : testData.generateTestData(dataSize, dimensions)) {
+            forest.update(point);
+        }
+
+        // Convert to an array of bytes and print the size
+
+        RandomCutForestMapper mapper = new RandomCutForestMapper();
+        mapper.setSaveExecutorContext(true);
+        mapper.setCopy(true);
+
+        Schema<RandomCutForestState> schema = RuntimeSchema.getSchema(RandomCutForestState.class);
+        LinkedBuffer buffer = LinkedBuffer.allocate(512);
+        byte[] bytes;
+        try {
+            RandomCutForestState state = mapper.toState(forest);
+            bytes = ProtostuffIOUtil.toByteArray(state, schema, buffer);
+        } finally {
+            buffer.clear();
+        }
+
+        System.out.printf("dimensions = %d, numberOfTrees = %d, sampleSize = %d, precision = %s%n", dimensions,
+                numberOfTrees, sampleSize, precision);
+        System.out.printf("protostuff size = %d bytes%n", bytes.length);
+
+        // Restore from protostuff and compare anomaly scores produced by the two
+        // forests
+
+        RandomCutForestState state2 = schema.newMessage();
+        ProtostuffIOUtil.mergeFrom(bytes, state2, schema);
+        RandomCutForest forest2 = mapper.toModel(state2);
+
+        double saveLambda = forest.getLambda();
+        forest.setLambda(10 * forest.getLambda());
+        forest2.setLambda(10 * forest2.getLambda());
+
+        for (int i = 0; i < numberOfTrees; i++) {
+            if (((CompactSampler) forest2.getComponents().get(i).getSampler())
+                    .getMaxSequenceIndex() != ((CompactSampler) forest.getComponents().get(i).getSampler())
+                            .getMaxSequenceIndex()) {
+                throw new IllegalStateException("Incorrect sampler state");
```

Comment:
This looks like something that should be checked in a unit test. I think if we want to show something in an example, it should be something like:
* sample for a while with one lambda value
* show that certain points are anomalous
* "flush" the sampler by increasing the Lambda
* update the model with a new distribution of points
* show that the point that was considered an anomaly before changing lambda is no longer an anomaly

Reply:
Maybe keep as a separate PR/issue? With the new refactoring of the samplers, all of the tests may need to be looked at.

Reply:
We can revise this later, but I think examples need to be clear about what they're an example of. Giving an example where dynamically changing the lambda produces an interesting result is good. I know that I started this with my serialization examples, but now I'm questioning whether we should include assertion-like statements.